pages/page/[page].tsx

```diff
@@ -25,25 +26,42 @@ export const getStaticProps: GetStaticProps = async (context) => {
   const parser = new Parser();
   let docs = cache.get<Doc[]>('docs');
   if (!docs) {
-    docs = (
-      await Promise.all(
-        channelsData.channels.map(async (channel, channelIndex) => {
-          const result = await parser.parseURL(channel.url);
-          return (
-            result?.items.map((item) => ({
-              ...item,
-              author: channel,
-              authorId: channelIndex
-            })) ?? []
-          );
-        })
-      )
-    ).flat();
-    docs.sort((a, b) => {
-      let da = new Date(b.pubDate as string);
-      let db = new Date(a.pubDate as string);
-      return +da - +db;
+    docs = [];
+    const docsArray: Array<Doc[]> = await Promise.all(
+      channelsData.channels.map(async (channel, channelIndex) => {
+        const result = await parser.parseURL(channel.url);
+        return (
+          result?.items.map((item) => ({
+            ...item,
+            author: channel,
+            authorId: channelIndex
+          })) ?? []
+        );
+      })
+    );
+    const maxHeap = new MaxHeap<[Doc, number, number]>((node1, node2) => {
```

Comment:
We better define an actual return type for heap item, as we gonna use it again in line 59.